client/trino-client/src/main/java/io/trino/client/OkHttpUtil.java

```diff
@@ -88,14 +91,30 @@ public static Interceptor basicAuth(String user, String password)
                 .build());
     }
 
-    public static Interceptor tokenAuth(String accessToken)
+    public static Interceptor tokenAuth(String accessToken, boolean followRedirects)
     {
         requireNonNull(accessToken, "accessToken is null");
         checkArgument(CharMatcher.inRange((char) 33, (char) 126).matchesAllOf(accessToken));
 
-        return chain -> chain.proceed(chain.request().newBuilder()
-                .addHeader(AUTHORIZATION, "Bearer " + accessToken)
-                .build());
+        return chain -> {
+            Request request = chain.request().newBuilder()
+                    .addHeader(AUTHORIZATION, "Bearer " + accessToken)
+                    .build();
+
+            Response response = chain.proceed(request);
+
```

Comment:
If `followRedirects` is enabled, then follow the redirect and reattach the `Authorization` header again while making the call.

Reply:
In Trino protocol, we sent the Authorization header with the request together. After received an redirect response, we should sent Authorization header in the redirect retry.
The `if redirect retry received 401 then retry with Authorization header` logic doesn't follow the current Trino protocol.
You might need to use [Network Interceptor](https://square.github.io/okhttp/features/interceptors/#network-interceptors).

Reply:
@oneonestar 
One of the documentation https://square.github.io/okhttp/features/interceptors/#choosing-between-application-and-network-interceptors says Network interceptors are not invoked for cached response.

```
Not invoked for cached responses that short-circuit the network.
```

Wouldn't this cause issues to redirects if anything serverd via `v1/*` has a `Cache-control` header with `max-age` field ?

Reply:
https://square.github.io/okhttp/features/caching/
> OkHttp implements an optional, off by default,

I don't see Trino client or server is using the cache by okhttp.